pymfe/itemset.py

```diff
@@ -0,0 +1,179 @@
+       value).
+
+    3. The initial assumption is that the user can change any optional
+       argument, without any previous verification of argument value or its
+       type, via kwargs argument of ``extract`` method of MFE class.
+
+    4. The return value of all feature extraction methods should be a single
+       value or a generic Sequence (preferably a :obj:`np.ndarray`) type with
+       numeric values.
+
+    There is another type of method adopted for automatic detection. It is
+    adopted the prefix ``precompute_`` for automatic detection of these
+    methods. These methods run while fitting some data into an MFE model
+    automatically, and their objective is to precompute some common value
+    shared between more than one feature extraction method. This strategy is a
+    trade-off between more system memory consumption and speeds up of feature
+    extraction. Their return value must always be a dictionary whose keys are
+    possible extra arguments for both feature extraction methods and other
+    precomputation methods. Note that there is a share of precomputed values
+    between all valid feature-extraction modules (e.g., ``class_freqs``
+    computed in module ``statistical`` can freely be used for any
+    precomputation or feature extraction method of module ``landmarking``).
+    """
+    @classmethod
+    def precompute_binary_matrix(cls,
+                                 C: np.ndarray,
+                                 **kwargs) -> t.Dict[str, t.Any]:
+        """Precompute the binary representation of attributes.
+
+        Parameters
+        ----------
+        C : :obj:`np.ndarray`, optional
+            Attributes from fitted data.
+
+        **kwargs
+            Additional arguments. May have previously precomputed before this
+            method from other precomputed methods, so they can help speed up
+            this precomputation.
+
+        Returns
+        -------
+        :obj:`dict`
+            With following precomputed items:
+                - ``itemset_binary_matrix`` (:obj:`list`): Binary
+                  representation of the attributes.
+        """
+        prepcomp_vals = {}
+        if C is not None and 'itemset_binary_matrix' not in kwargs:
+            itemset_binary_matrix = MFEItemset._matrix_to_binary(C)
+            prepcomp_vals["itemset_binary_matrix"] = itemset_binary_matrix
+        return prepcomp_vals
+
+    @staticmethod
+    def _array_to_binary(array: np.ndarray):
+        """Convert an array to its binary representation."""
+        values = np.unique(array)
+        res = np.zeros((array.shape[0], values.shape[0])).astype(bool)
+        for i, val in enumerate(values):
+            res[:, i] = array == val
+        return res
+
+    @staticmethod
+    def _matrix_to_binary(C: np.array):
+        """Convert an matrix to its binary representation."""
+        return [MFEItemset._array_to_binary(col) for col in C.T]
+
+    @classmethod
+    def ft_two_itemset(cls,
```

Comment:
The metafeature name says "two_itemset" metafeature but the pydoc says "one itemset metafeature".

Reply:
Yes, it is wrong. I will solve it. Thank You.